lib/global-styles.php

```diff
@@ -242,6 +242,64 @@ function gutenberg_experimental_global_styles_enqueue_assets_editor() {
 	}
 }
 
+/**
+ * Converts the block implicit attributes into CSS declarations.
+ *
+ * @return string
+ */
+function gutenberg_experimental_global_styles_output_implicit_attributes( $block_attributes ) {
+	$declaration = '';
+	foreach ( $block_attributes as $att ) {
+		$declaration .= $att . ': var(--wp-' . $att . ');';
+	}
+	return $declaration;
+}
+
+/**
+ * Returns the CSS with the block style hooks
+ *
+ * @return string
+ */
+function gutenberg_experimental_global_styles_block_style_hooks() {
+	// This data should come from the block.json.
+	// We'd need the ability to have many selectors per block
+	// like for heading (core/heading/h1, core/heading/h2, etc)
+	// but potentially for other blocks that have complex layouts
+	// (ex: for gallery we may want to target the wrapper element
+	// and the individual images).
+	//
+	// Having access to this via the block registry
+	// requires block server registration.
+	$block_data = array(
+		'core/paragraph'  => array(
+			'supports' => array( 'line-height', 'font-size', 'color' ),
+			'selector' => 'p',
+		),
+		'core/heading'    => array(
+			'supports' => array( 'line-height', 'font-size', 'color' ),
+			'selector' => 'h1, h2, h3, h4, h5, h6',
+		),
+		'core/columns'    => array(
+			'supports' => array( 'color' ),
+			'selector' => '.wp-block-columns',
```

Comment:
Would it be ok to use class selectors as a default and not provide the selectors explicitly?